compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -826,14 +826,19 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
      */
     def becomes(rhs: Tree)(implicit ctx: Context): Tree =
       if (tree.symbol is Method) {
-        val setr = tree match {
-          case Ident(_) =>
-            val setter = tree.symbol.setter
-            assert(setter.exists, tree.symbol.showLocated)
-            ref(tree.symbol.setter)
-          case Select(qual, _) => qual.select(tree.symbol.setter)
+        val setter = tree.symbol.setter
+        assert(setter.exists, tree.symbol.showLocated)
+        val qual = tree match {
+          case id: Ident =>
+            id.tpe match {
```

Comment:
There's already three versions of `desugarIdent` in the codebase. I suggest we try to replace them by a common version in tpd.scala or something like that instead of adding one more variation.

Reply:
I am only interested in the prefix of the desugared ident. Should I still extract the functionality and do something like:
```scala
val qual = tree match {
  case id: Ident =>
    desugarIdent(id).asInstanceOf[Select].qual
  ...
```
This would create an unnecessary `Select` node

Reply:
we could have a `desugarIdentPrefix` method.